### ISSUE TITLE
No images shown in full record display by default

### ISSUE THREAD

**Issue by mephillips-durham** [projectblacklight/spotlight#2992]
In a freshly-bootstrapped rails application based on the Spotlight engine, any images, whether static or IIIF, are only shown in the brief results screens and do not appear in the full record display. This is not helpful to new users, who may struggle to work out how to add the images. Most users will want images displayed on the full record view.

By contrast, when you Edit a record, clicking the Edit button from the full record display, the images are shown. It appears this is because the edit screens use an older mechanism. I wonder whether the images disappeared from the default full record display when Blacklight moved to using View Controllers?

I don't know whether there is a very simple fix that I have not found, and I don't know whether it needs addressing in Spotlight or in Blacklight.

### Background investigations
If I turn on diagnostics in the rendering so that the HTML is marked up with the template filenames using
```
config.action_view.annotate_rendered_view_with_filenames = true
```
then in the normal full record display we have these templates in use:

```
/home/vagrant/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/blacklight-7.34.0/app/components/blacklight/document_component.rb
/home/vagrant/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/blacklight-7.34.0/app/components/blacklight/document_metadata_component.rb
/home/vagrant/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/blacklight-7.34.0/app/components/blacklight/metadata_field_component.rb
```

The metadata_field_component would be repeated for each metadata field in the record, displaying each in a new row with the field name on the left. I would expect the document_component.rb to generate or include something to make the image appear. It does make a couple of lines appear with "og" metadata for use by Facebook etc.:

```
<meta property="og:title" content="Example title here" />
<meta property="og:og:image" content="/images/535-8c9b943d3fccdc21ebd2930481a0f3c6/full/!400,400/0/default.jpg" />
```

But there is no <iframe> or <img> tag to make the image appear.

By contrast, if we look at the templates used in the Edit view of the record, we get these:

```
/home/vagrant/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/bundler/gems/spotlight-0f5a47eefe43/app/views/spotlight/catalog/edit.html.erb
/home/vagrant/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/blacklight-7.34.0/app/views/catalog/_show_header.html.erb
app/views/catalog/_openseadragon_default.html.erb
```
and others which follow to make the form elements appear. The _openseadragon_default.html.erb template makes the IIIF viewer appear.

On our previous attempt to implement Spotlight, in August 2021, we had replaced this with Mirador, and that was good for both the editing view and the full record view. Repeating that modification with the latest Spotlight and Blacklight now only affects the editing view, and the full record display has no image viewer, neither Mirador nor OpenSeadragon.

**Comment by SunnyMoonSchott** [comment]
The following is a reproduction of this issue using a Durham University [IIIF manifest](https://iiif.durham.ac.uk/manifests/trifle/museum/object/00/44/00446/manifest).

As you can see, the preview images are not rendered in the items tab:

![Screenshot from 2024-03-15 13-58-40](https://github.com/projectblacklight/spotlight/assets/18484983/3e496d9f-f737-4276-beab-f033a77d1b89)

Not rendered in the individual item view:

![Screenshot from 2024-03-15 13-58-52](https://github.com/projectblacklight/spotlight/assets/18484983/743b9f4a-b360-489b-a4be-923865284933)

Or in the item edit view:

![Screenshot from 2024-03-15 13-59-25](https://github.com/projectblacklight/spotlight/assets/18484983/03fc8509-9739-42ab-b78e-80c986c5b07f)

The pictures are viewable as thumbnails when creating a static slideshow:

![Screenshot from 2024-03-15 14-03-01](https://github.com/projectblacklight/spotlight/assets/18484983/ab1dca60-2df5-49c4-bf29-88fcdf114256)

And are displayed properly in the Feature Page when displayed as a static slideshow:

![Screenshot from 2024-03-15 14-03-11](https://github.com/projectblacklight/spotlight/assets/18484983/cbe1f35b-3238-4a65-a4fe-2e3ee74f846d)

However, when displayed in a viewer window, the image is not correctly rendered:

![Screenshot from 2024-03-15 14-42-23](https://github.com/projectblacklight/spotlight/assets/18484983/993ea1c8-4e20-4b90-84c2-71c7db18d391)

Inspecting the elements shows that the viewer is not rendered at all:

![Screenshot from 2024-03-15 14-45-51](https://github.com/projectblacklight/spotlight/assets/18484983/dba3ba6a-9d3e-480c-8066-084a916687e9)

This spotlight instance was instantiated on an ubuntu vagrant box using the following script:

`#!/bin/bash

echo "Start of bundle-install.sh"

# Spotlight
cd
if [ ! -e dur-spotlight/app ]; then
    echo "Spotlight bundle install"

    # Need these lines for initial provisioning, as parent environment has not yet had these executed
    source $HOME/.asdf/asdf.sh
    source $HOME/.asdf/completions/asdf.bash

    # Create new Spotlight application
    SKIP_TRANSLATION=1 rails new dur-spotlight --quiet -m https://raw.githubusercontent.com/projectblacklight/spotlight/main/template.rb
    cd dur-spotlight
    SKIP_TRANSLATION=1 rake db:migrate

    # Make corrections to generated manifest.js
    cd
    sed -i 's/js\/\//js\n\/\//' dur-spotlight/app/assets/config/manifest.js
    sed -i 's/^\/\/= link application.js$//' dur-spotlight/app/assets/config/manifest.js
    
    cd
    cat <<EOF >> dur-spotlight/.solr_wrapper.yml
  # Added by bundle-install.sh to persist data
  persist: true
# Added by bundle-install.sh to avoid data being kept in /tmp
download_dir: /home/vagrant/tmp
instance_dir: /home/vagrant/solr_data
# Added by bundle-install.sh to listen on all addresses
env:
  SOLR_JETTY_HOST: 0.0.0.0
EOF
fi

echo "End of bundle-install.sh"
`

**Comment by corylown** [comment]
This is fixed by version v0.7.0 of openseadragon-rails: https://rubygems.org/gems/openseadragon/versions/0.7.0
